File: LeetCode/Python/Medium/2125-Number-of-Laser-Beams-in-a-Bank/solution.py
class Solution:
    def numberOfBeams(self, bank: List[str]) -> int:
        result=0

        ones_in_previous_row=0

        for row in bank:
            ones=row.count('1')

            if ones==0:
                continue

            product=ones_in_previous_row * ones

            result+=product

            ones_in_previous_row=ones

        return result

File: LeetCode/Rust/Medium/2125-Number-of-Laser-Beams-in-a-Bank/solution.rs
impl Solution
{
    pub fn number_of_beams(bank: Vec<String>) -> i32
    {
        let mut result: i32=0;

        let mut ones_in_previous_row: i32=0;

        for row in bank.iter()
        {
            let ones: i32=row.chars().filter(|&digit| digit=='1').count() as i32;

            if ones==0
            {
                continue;
            }

            let product: i32=ones * ones_in_previous_row;

            result+=product;

            ones_in_previous_row=ones;
        }

        result
    }
}